### PR TITLE
[SM-198] Empty Secrets View

### PR DESCRIPTION
**Pull request by Thomas-Avery**
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

This PR is for creating a view for when no secrets are available. 

## Code changes

- apps/web/src/locales/en/messages.json

Adding new messages for empty secrets view

- bitwarden_license/bit-web/src/app/sm/secrets/secrets-list.component.html

Display empty secrets view if no secrets are passed in as input.

- bitwarden_license/bit-web/src/app/sm/secrets/secrets-list.component.ts

Add event emitters for when buttons are clicked for empty secret view.

- bitwarden_license/bit-web/src/app/sm/secrets/secrets.component.html
bitwarden_license/bit-web/src/app/sm/secrets/secrets.component.ts

Handle launching new secret dialog

## Screenshots

image

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: apps/web/src/locales/en/messages.json [hunk 1]
```diff
@@ -5409,6 +5409,18 @@
...
+    "message":"New Secret"
+  },
+  "importSecrets":{
+    "message":"Import Secrets"
+  },
+  "emptyListTitle":{
+    "message":"Nothing to show yet"
+  },
+  "emptyListMessage":{
+    "message": "Add a new secret or import secrets to get started"
```
Comment (Hinton) [inline apps/web/src/locales/en/messages.json:5422]: This is secrets specific and we should prefix the localization string with secrets.

File: bitwarden_license/bit-web/src/app/sm/secrets/secrets-list.component.html [hunk 2]
```diff
@@ -1,4 +1,24 @@
-<bit-table>
+<div *ngIf="secrets.length == 0" class="no-items tw-justify-start tw-pt-6">
+  <img class="no-items-image" aria-hidden="true" />
```
Comment (Hinton) [inline bitwarden_license/bit-web/src/app/sm/secrets/secrets-list.component.html:2]: Thoughts on moving the `no-items-image` out of the wrapped `no-items`, so we can implement this using tailwind? I would like to remove the `no-items` class long term.

File: bitwarden_license/bit-web/src/app/sm/secrets/secrets-list.component.html [hunk 3]
```diff
@@ -1,4 +1,24 @@
...
+  <div class="tw-max-w-xs">
+    <h3 class="tw-font-semibold">
+      {{ "emptyListTitle" | i18n }}
+    </h3>
+    <p>{{ "emptyListMessage" | i18n }}</p>
+  </div>
+  <div class="tw-space-x-2">
+    <button bitButton buttonType="secondary" (click)="this.openNewSecretDialogEvent.emit()">
```
Comment (Hinton) [inline bitwarden_license/bit-web/src/app/sm/secrets/secrets-list.component.html:10]: We don't need to prefix these with `this` I believe.

File: bitwarden_license/bit-web/src/app/sm/secrets/secrets-list.component.ts [hunk 4]
```diff
@@ -17,6 +17,8 @@ export class SecretsListComponent implements OnDestroy {
   @Output() projectsEvent = new EventEmitter<string>();
   @Output() deleteSecretEvent = new EventEmitter<string>();
   @Output() onSecretCheckedEvent = new EventEmitter<string[]>();
+  @Output() openNewSecretDialogEvent = new EventEmitter();
+  @Output() openImportSecretsDialogEvent = new EventEmitter();
```
Comment (Hinton) [inline bitwarden_license/bit-web/src/app/sm/secrets/secrets-list.component.ts:21]:
```suggestion
  @Output() newSecretEvent = new EventEmitter();
  @Output() importSecretsEvent = new EventEmitter();
```

Comment (Thomas-Avery) [reply bitwarden_license/bit-web/src/app/sm/secrets/secrets-list.component.ts]: Resovled in https://github.com/bitwarden/clients/pull/3585/commits/a423a9c74d6fd6e1011226cda0f26f5b2e02545b

File: bitwarden_license/bit-web/src/app/sm/secrets/secrets-list.component.html (hunk 3 above)
Comment (Thomas-Avery) [reply bitwarden_license/bit-web/src/app/sm/secrets/secrets-list.component.html]: Resolved in https://github.com/bitwarden/clients/pull/3585/commits/a423a9c74d6fd6e1011226cda0f26f5b2e02545b

File: bitwarden_license/bit-web/src/app/sm/secrets/secrets-list.component.html (hunk 2 above)
Comment (Thomas-Avery) [reply bitwarden_license/bit-web/src/app/sm/secrets/secrets-list.component.html]: Unwrapped `no-items-image` from the `no-items` container in apps/web/src/scss/base.scss

Resolved in https://github.com/bitwarden/clients/commit/a423a9c74d6fd6e1011226cda0f26f5b2e02545b

File: apps/web/src/locales/en/messages.json (hunk 1 above)
Comment (Thomas-Avery) [reply apps/web/src/locales/en/messages.json:5422]: Resolved in https://github.com/bitwarden/clients/commit/a423a9c74d6fd6e1011226cda0f26f5b2e02545b

**Review by Hinton — APPROVED** [review]

**Pull request merged**